docs/api/proposed_endpoints.md

```diff
@@ -0,0 +1,186 @@
+[TOC]
+
+# Posts related actions
+
+## Create a post
+
+Create a post of contents.
+
+```
+POST api/posts
+```
+
+**Paramaters**
+
+Name         | Description
+-------------|--------------------------------------
+id           | identity of the post
+posted_by    | user who is posting the content
+posted_at    | social media ploatform to post
+posted_time  | time at which post should be posted
+content_type | type of content, text or image
+content      | content of the post
+image        | image in the post
+is_approved  | whether the post is approved or not
+is_posted    | whether the post is posted or not
+
+**Request**
+```json
+{
+    "id": "724",
+    "posted_by": "John Micky",
+    "posted_at": "Twitter",
+    "posted_time": "",
+    "content_type": "text",
+    "content": "Anything that user has to write",
+    "image": "",
+    "is_approved": "False",
+    "is_posted": "False"
+}
+```
+
+**Response**
+```json
+
+Status: 201 Created
+{
+    "id": "724",
+    "posted_by": "John Micky",
+    "posted_at": "Twitter",
+    "posted_time": "",
+    "content_type": "text",
+    "content": "Anything that user has to write",
+    "image": "http://xyz.com/path/of/image.jpg",
+    "is_approved": "False",
+    "is_posted": "False"
+
+}
+```
+
+## Update a post
+
+```
+PATCH api/posts/:post_id (Requires Authorization)
+```
+
+**Parameters**
+
+Name         | Description
+-------------|--------------------------------------
+new_content  | new content of the post
+new_image    | new image in the post
+posted_time  | time at which post should be posted
+content_type | current content type of the post
+
+**Request**
+```json
+{
+    "new_content": "",
+    "new_image":"http://xyz.com/path/of/image.jpg",
+    "posted_time": "2018-10-10T13:43:42Z",
+    "content_type": "image"
+}
+```
+
+**Response**
+```json
+
+Status: 200 OK
+{
+    "id": "724",
+    "posted_by": "John Micky",
+    "posted_at": "Twitter",
+    "posted_time": "2018-10-10T13:43:42Z",
+    "content_type": "image",
```

Comment:
What does `content_type` refers to? What values can it get?

Reply:
We are really sorry about this. Because when you asked to add one more column about `data type` we thought by mistake that you are asking about adding one more parameter like `content_type`. But now we understood that you were asking to add one more column in the `parameters` table to describe the `data type` of those parameters.

Reply:
Ack